File: f#/words/lib/Numerical.fs
namespace lib.Numerical

module Numerical =
  let fibonacci n =
    let rec fibonacciCalc n acc2 acc1 =
      match n with
        | n when n < 0 -> -1
        | n when n <= 1 -> n
        | n when n = 2 -> acc1 + acc2
        | _ -> fibonacciCalc (n - 1) acc1 (acc1 + acc2)
    fibonacciCalc n 0 1

  let factorial n = 
    let rec factorialCalc n acc =
      match n with
        | n when n < 0 -> -1
        | n when n <= 1 -> acc
        | _ -> factorialCalc (n - 1) (n * acc)
    factorialCalc n 1

  let rec gcd a b =
    if b = 0 then
      a
    else
      gcd b (a % b)

  let lcm a b =
    (a / (gcd a b)) * b

File: _experimental/_kotlin/words/lib/src/main/kotlin/org/example/Numerical.kt
object Numerical {
	fun fibonacci(n: Int): Int {
		if (n < 0) return -1
		if (n < 2) return n
		var acc1 = 1
		var acc2 = 0
		for (i in 3..n) {
			val temp = acc1
			acc1 = acc1 + acc2
			acc2 = temp
		}
		return acc1 + acc2
	}

	fun factorial(n: Int): Int {
		if (n < 0) return -1
		var acc = 1
		for (i in 2..n) {
			acc *= i
		}
		return acc
	}

	fun gcd(a: Int, b: Int): Int {
		if (b == 0) return a
		return gcd(b, a % b)
	}

	fun lcm(a: Int, b: Int): Int {
		return (a / gcd(a, b)) * b
	}
}